project/src/components/videoplayer/videoPlayer.tsx

```diff
@@ -0,0 +1,37 @@
+import {useEffect, useRef} from 'react';
+
+type VideoPlayerProps = {
+  previewVideoLink: string;
+  previewImage: string;
+  isPlaying: boolean;
+}
+
+function VideoPlayer(props: VideoPlayerProps): JSX.Element {
+  const videoRef = useRef<HTMLVideoElement>(null);
+  const {previewImage, previewVideoLink, isPlaying} = props;
+
+  useEffect(() => {
+    const timeOut = setTimeout(() => {
+      if (isPlaying && videoRef.current) {
+        videoRef.current.play();
+      }
+    }, 1000);
+
+    return () => {
+      if (isPlaying && videoRef.current) {
+        videoRef.current.pause();
+        videoRef.current.src = '';
+        videoRef.current.src = previewVideoLink;
+        if (timeOut)  {
+          clearTimeout(timeOut);
+        }
+      }
+    };
+  });
+
+  return (
+    <video src={previewVideoLink} poster={previewImage} ref={videoRef} width="280" height="175" muted/>
+  );
+}
```

Comment:
Немного упростил:
```suggestion
function VideoPlayer(props: VideoPlayerProps): JSX.Element {
  const videoRef = useRef<HTMLVideoElement>(null);
  const {previewImage, previewVideoLink, isPlaying} = props;

  useEffect(() => {
    const video = videoRef.current;
    const timeout = setTimeout(() => {
      if (isPlaying && video) {
        video.play();
      }
    }, 1000);

    return () => {
      if (isPlaying && video) {
        video.src = previewVideoLink;
      }

      clearTimeout(timeout);
    };
  }, [isPlaying, previewVideoLink]);

  return (
    <video className='videoplayer' src={previewVideoLink} poster={previewImage} ref={videoRef} muted/>
  );
}
```

Стили можно поместить в `videoplayer.css` и заимпортить через `import './videoplayer.css';`

```css
.videoplayer {
  width: 280px;
  height: 175px;
  object-fit: cover;
}
```